File: backend/Domain/Models/Table.cs
using System.Text.Json.Serialization;
using Domain.DataBaseAccess;
using Domain.Logic;

namespace Domain.Models;

public class Table : ITableDataBase
{
    public long? Id { get; set; }
    [JsonPropertyName("restaurant_id")] public long RestaurantId { get; init; }
    public int Number { get; init; }
    public string Token { get; set; }
    [JsonPropertyName("qr_code_image_path")] public string QrCodeImagePath { get; set; }

    public Table(){}
    public Table(long? id, long restaurantId, int number)
    {
        Id = id;
        RestaurantId = restaurantId;
        Number = number;
        GenerateTokenAndQrCode();
    }

    private void GenerateTokenAndQrCode()
    {
        Token = TokenEncryptor.GenerateToken(Id, RestaurantId);
        QrCodeImagePath = TableQrCode.GenerateAndSaveQrCode(Number, Token);
    }
}

File: backend/API/PY_bots/Model/Table.py
from pydantic import BaseModel
from typing import Optional


class Table(BaseModel):
    id: Optional[int]
    restaurant_id: int
    number: int
    token: Optional[str]
    qr_code_image_path: Optional[str]

    def __init__(self,
                 restaurant_id: int,
                 number: int,
                 qr_code_image_path: Optional[str] = None,
                 token: Optional[str] = None,
                 id: Optional[int] = None):
        super().__init__(
            id=id,
            restaurant_id=restaurant_id,
            number=number,
            qr_code_image_path=qr_code_image_path,
            token=token
        )
